### PR TITLE
Add new features in config editor

### PR DESCRIPTION
**Pull request by kanechen66**
## Description

In this PR, we added the below new features in config editor
1. UI menu to read/write variable from SUT in runtime
2. UI menu to delete variable listed in platformcfg.xml on SUT in runtime
3. UI to show, admin mode,  manufacturing mode and mfci policy on the lower right side
4. Status box show log output from config editor so that user can know what vl or xml file is loaded
image

For details on how to complete these options and their meaning refer to [CONTRIBUTING.md](https://github.com/microsoft/mu/blob/HEAD/CONTRIBUTING.md).

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [x] Includes tests?
- [ ] Includes documentation?

## How This Was Tested
We tested on real SUT and verify each function we added. 
<_Describe the test(s) that were run to verify the changes._>

## Integration Instructions

<_Describe how these changes should be integrated. Use N/A if nothing is required._>
N/A

### CODE REVIEW

**Comment by kanechen66** [comment]
@apop5 
Hi Aaron
pls help to review.
thank you :)

**Comment by kanechen66** [comment]
hi all, 
Do you have any feedback for this PR?
thanks

File: SetupDataPkg/Tools/ConfigEditor.py [hunk 1]
```diff
@@ -784,6 +850,34 @@ def load_from_delta(self):
...
+        self.output_current_status("Settings are set to system and save to RuntimeVar.vl")
+
+    def del_all_variable_runtime(self):
+        if (not ask_yes_no(f"Do you want to delete all variables in {self.config_xml_path} on system?\n")):
+            return
+
+        schema = Schema.load(self.config_xml_path)
+        for knob in schema.knobs:
+            self.output_current_status(f"Delete variable {knob.name} with namespace {knob.namespace}")
+            uefi_var_write.delete_var_by_guid_name(knob.name, knob.namespace)
```
Comment (apop5) [inline SetupDataPkg/Tools/ConfigEditor.py:878]:
```suggestion
            rc = uefi_var_write.delete_var_by_guid_name(knob.name, knob.namespace)
            if rc == 0:
                self.output_current_status(f"{knob.name} variable was not deleted from system {rc}")
            else:
                self.output_current_status(f"{knob.name} variable is deleted from system")```

Comment (apop5) [reply SetupDataPkg/Tools/ConfigEditor.py:878]: we want to print the output results accurately, if the variables dont' exist, we want that reflected in the logs.

File: SetupDataPkg/Tools/ConfigEditor.py [hunk 2]
```diff
@@ -863,6 +959,11 @@ def load_cfg_file(self, path, file_id, clear_config):
         for menu in self.menu_string:
             self.file_menu.entryconfig(menu, state="normal")
 
+        for menu in self.variable_menu_string:
+            self.variable_menu.entryconfig(menu, state="normal")
```
Comment (apop5) [inline SetupDataPkg/Tools/ConfigEditor.py:963]:
```suggestion
        if self.admin_mode:
            for menu in self.variable_menu_string:
                self.variable_menu.entryconfig(menu, state="normal")
```



In the case that the editor was not started in admin mode, then it should not attempt to make the variable menu entries visible.

**Review by os-d — APPROVED** [review]

File: SetupDataPkg/Tools/ConfigEditor.py (hunk 2 above)
Comment (kanechen66) [reply SetupDataPkg/Tools/ConfigEditor.py]: addressed in 
https://github.com/microsoft/mu_feature_config/pull/413/commits/e6f697b5cb45f996a49bfe3faaf16de4b177f7f0

File: SetupDataPkg/Tools/ConfigEditor.py (hunk 1 above)
Comment (kanechen66) [reply SetupDataPkg/Tools/ConfigEditor.py]: addressed in 
https://github.com/microsoft/mu_feature_config/pull/413/commits/e6f697b5cb45f996a49bfe3faaf16de4b177f7f0

**Comment by kanechen66** [comment]
> @kanechen66 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

**Comment by kanechen66** [comment]
hi all 
I've addressed the comments. Also do a small tweak in UI when config var is empty in https://github.com/microsoft/mu_feature_config/commit/d8aba79e8874f8fa5093456519b9fbddaa36b15a  
thank you :)

**Comment by apop5** [comment]
It looked good in the testing of the latest.

**Pull request merged**